### PR TITLE
feat(comment): Avoid parsing javadoc when it isn't necessary.

### LINKED ISSUE #666: put comments as ctstatement only if -j flag is present
Discussion for @arnobl use case

Proposal: widen the scope of -j flag as follows:
put comments as ctstatement only if -j flag is present
and update the documentation of the -j flag:
-j consider documentation:
this flags enables the documentation handling: adds statement comments in the model, rewrite comments on pretty-printing

### PR DESCRIPTION
**Pull request by GerardPaligot**
Fixes for a diabolical issue! :fire:

Closes #666

### CODE REVIEW

**Comment by monperrus** [comment]
could you update the command line and website documentation as well?

**Comment by GerardPaligot** [comment]
> command line

This is the command line of this feature:

```java
// Enable generation of javadoc.
sw1 = new Switch("generate-javadoc");
sw1.setShortFlag('j');
sw1.setLongFlag("generate-javadoc");
sw1.setHelp("Enable the generation of the javadoc.");
sw1.setDefault("false");
jsap.registerParameter(sw1);
```

We should deprecate this argument and create a new one.

> website documentation

Which part?

**Comment by GerardPaligot** [comment]
AppVayor and Travis seems down... but PR updated.

File: doc/command_line.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 title: Command Line
 tags: [usage]
 keywords: command, line, usage, java, jar
-last_updated: October 1, 2015
+last_updated: May 25, 2015
```
Comment (monperrus) [inline doc/command_line.md]: 2016

File: doc/command_line.md [hunk 2]
```diff
@@ -130,5 +119,13 @@ Options :
         Disable the copy of resources from source to destination folder.
 
   [-j|--generate-javadoc]
-        Enable the generation of the javadoc.
+        Enable the generation of the javadoc. Deprecated, use enable-comments
+        argument.
+
+  [-c|--enable-comments]
+        Adds all comments of the target source code in the Spoon AST.
```
Comment (monperrus) [inline doc/command_line.md]: "Adds all code comments in the Spoon AST (Javadoc, line-based comments), rewrites them when pretty-printing"

**Comment by GerardPaligot** [comment]
@monperrus PR ok.

**Pull request merged**